src/components/useList/components/ListItemView/ListItemView.tsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+
+import {Check, ChevronDown, ChevronUp} from '@gravity-ui/icons';
+
+import {Icon} from '../../../Icon';
+import {Text, colorText} from '../../../Text';
+import {Flex, FlexProps, spacing} from '../../../layout';
+import type {QAProps} from '../../../types';
+import {block} from '../../../utils/cn';
+import {LIST_ITEM_DATA_ATR, modToHeight} from '../../constants';
+import type {ListItemId, ListItemSizeType} from '../../types';
+
+import './ListItemView.scss';
+
+const b = block('list-item-view');
+
+export interface ListItemViewProps extends QAProps {
+    /**
+     * Ability to override default html tag
+     */
+    as?: keyof JSX.IntrinsicElements;
+    /**
+     * @default `m`
+     */
+    size?: ListItemSizeType;
+    height?: number;
+    selected?: boolean;
+    active?: boolean;
+    disabled?: boolean;
+    /**
+     * By default hovered elements has active styles. You can disable this behavior
+     */
+    activeOnHover?: boolean;
+    /**
+     * Build in indentation component to render nested views structure
+     */
+    indentation?: number;
+    /**
+     * Show selected icon if selected and reserve space for this icon
+     */
+    hasSelectionIcon?: boolean;
+    /**
+     * Note: if passed and `disabled` option is `true` click will not be appear
+     */
+    onClick?(): void;
+    style?: React.CSSProperties;
+    title: React.ReactNode;
+    subtitle?: React.ReactNode;
+    startSlot?: React.ReactNode;
+    endSlot?: React.ReactNode;
+    className?: string;
+    role?: React.AriaRole;
+    expanded?: boolean;
+    /**
+     * `[${LIST_ITEM_DATA_ATR}="${id}"]` data attribute to find element.
+     * For example for scroll to
+     */
+    id: ListItemId;
+}
+
+interface SlotProps extends FlexProps {
+    indentation?: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
+}
+
+export const Slot = ({children, indentation: indent = 1, className, ...props}: SlotProps) => {
+    return (
+        <Flex className={b('slot', {indent}, className)} {...props}>
+            {children}
+        </Flex>
+    );
+};
+
+const renderSafeIndentation = (indentation?: number) => {
+    if (indentation && indentation >= 1 && indentation < 11) {
+        return <Slot indentation={Math.floor(indentation) as SlotProps['indentation']} />;
+    }
+    return null;
+};
+
+export const ListItemView = React.forwardRef(
+    (
+        {
+            id,
+            as = 'div',
+            size = 'm',
+            active,
+            selected,
+            disabled,
+            activeOnHover = true,
+            className,
+            hasSelectionIcon = true,
+            indentation,
+            startSlot,
+            subtitle,
+            endSlot,
+            title,
+            height,
+            expanded,
+            style,
+            role = 'listitem',
```

Comment:
role `option` more suitable here, it supports `aria-selected` attribute, while `listitem` doesn't